manageiq-operator/api/v1alpha1/helpers/miq-components/kafka/kafka.go

```diff
@@ -0,0 +1,351 @@
+						"capabilities": map[string]interface{}{
+							"drop": []string{"ALL"},
+						},
+						"privileged":             false,
+						"readOnlyRootFilesystem": false,
+						"runAsNonRoot":           true,
+					},
+				},
+				"userOperatorContainer": map[string]interface{}{
+					"securityContext": map[string]interface{}{
+						"allowPrivilegeEscalation": false,
+						"capabilities": map[string]interface{}{
+							"drop": []string{"ALL"},
+						},
+						"privileged":             false,
+						"readOnlyRootFilesystem": false,
+						"runAsNonRoot":           true,
+					},
+				},
+				"tlsSidecarContainer": map[string]interface{}{
+					"securityContext": map[string]interface{}{
+						"allowPrivilegeEscalation": false,
+						"capabilities": map[string]interface{}{
+							"drop": []string{"ALL"},
+						},
+						"privileged":             false,
+						"readOnlyRootFilesystem": false,
+						"runAsNonRoot":           true,
+					},
+				},
+			},
+			"tlsSidecar": map[string]interface{}{
+				"resources": map[string]interface{}{
+					"requests": map[string]interface{}{
+						"cpu":    "500m",
+						"memory": "128Mi",
+					},
+					"limits": map[string]interface{}{
+						"cpu":    "500m",
+						"memory": "128Mi",
+					},
+				},
+			},
+			"userOperator": map[string]interface{}{
+				"resources": map[string]interface{}{
+					"requests": map[string]interface{}{
+						"cpu":    1,
+						"memory": "1Gi",
+					},
+					"limits": map[string]interface{}{
+						"cpu":    1,
+						"memory": "1Gi",
+					},
+				},
+			},
+			"topicOperator": map[string]interface{}{
+				"resources": map[string]interface{}{
+					"requests": map[string]interface{}{
+						"cpu":    1,
+						"memory": "1Gi",
+					},
+					"limits": map[string]interface{}{
+						"cpu":    1,
+						"memory": "1Gi",
+					},
+				},
+			},
+		},
+		// "clusterCa": map[string]interface{}{
+		// 	"generateCertificateAuthority": false,
+		// },
+	}
+}
+
+func KafkaCluster(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*unstructured.Unstructured, controllerutil.MutateFn) {
+	kafkaClusterCR := &unstructured.Unstructured{}
+	kafkaClusterCR.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "kafka.strimzi.io",
+		Kind:    "Kafka",
+		Version: "v1beta2",
+	})
+	kafkaClusterCR.SetName("manageiq")
+	kafkaClusterCR.SetNamespace(cr.Namespace)
+
+	kafkaCRSpec := KafkaClusterSpec()
+
+	if cr.Spec.StorageClassName != "" {
+		kafkaStorage := kafkaCRSpec["kafka"].(map[string]interface{})["storage"].(map[string]interface{})
+		kafkaStorage["class"] = cr.Spec.StorageClassName
+		zookeeperStorage := kafkaCRSpec["zookeeper"].(map[string]interface{})["storage"].(map[string]interface{})
+		zookeeperStorage["class"] = cr.Spec.StorageClassName
+	}
+
+	kafkaResourceRequests := kafkaCRSpec["kafka"].(map[string]interface{})["resources"].(map[string]interface{})["requests"].(map[string]interface{})
+	kafkaResourceRequests["memory"] = "1Gi"
+	kafkaResourceRequests["cpu"] = "200m"
```

Comment:
@bdunne Is there a reason why we don't use default values for `cr.Spec.KafkaMemoryRequest` etc in upstream? I was trying to use those values here but its not in the CR spec